### ISSUE TITLE
Confusion with fastcall and regparm

### ISSUE THREAD

**Issue by yrp604** [Vector35/binaryninja-api#1741]
To be explicit, I'm using these definitions for fastcall and regparam:
fastcall - ecx, edx, stack
regparm(3) - eax, edx, ecx, stack

These match these docs:
https://docs.microsoft.com/en-us/cpp/cpp/fastcall?view=vs-2019
https://gcc.gnu.org/onlinedocs/gcc/x86-Function-Attributes.html
https://clang.llvm.org/docs/AttributeReference.html#regparm
https://en.wikipedia.org/wiki/X86_calling_conventions#List_of_x86_calling_conventions

I have a regparm(3) function:
![image](https://user-images.githubusercontent.com/8716677/83383249-99919480-a399-11ea-8c6c-9f7c53992f19.png)

Using the above definitions, this correctly detected as regparm(3) (see the argument list in the function body), but is listed in its signature as __fastcall. So ok, perhaps binja uses a different definition of fastcall?

I can change it's type signature to `int32_t __convention("regparm")  dummyfs_mount(int32_t arg1, int32_t arg2, int32_t arg3, int32_t arg4)`. This doesn't result in a display change, which kind of makes sense if __fastcall and regparm are aliases. Where it really gets weird is when I change it to use `__convention("fastcall")` it _does_ change.

![image](https://user-images.githubusercontent.com/8716677/83383585-4b30c580-a39a-11ea-87be-73c65e214a78.png)

This means __fastcall and __convention("fastcall") are different, which seems really weird? And the definition binja is using for __fastcall is different than gcc's, clang's, and msv'sc?

[dmyfs.ko.txt](https://github.com/Vector35/binaryninja-api/files/4709864/dmyfs.ko.txt)

**Comment by yrp604** [comment]
Oh, duh:
```
>>> current_function.calling_convention
<calling convention: x86 regparm>
```

So yeah, for some reason regparm is displaying as __fastcall in the signature.

**Comment by plafosse** [comment]
Fixed in >= 2.4.2998
